add more tips for PR reviews (#3)
pixielabs/handbook: 03-project-delivery/06-reviews.md

- Put up a draft PR to get early feedback, it prevents us going down rabbit holes.
- Provide context about why the work has been done.
  - Including a link to the ticket is good.
  - Pulling out relevant tasks or ACs into the description is better.
- Explain how to run code if needed.
  - Explain how to set up, steps to take when testing and the expected behaviour.
- Take feedback on board.
  - Don’t be too defensive of the code, we're a team and collaboration
  makes our code better.
- Assume the reviewer's tone is a friendly/ curious one.
  - Written feedback is often read in an overly critical tone in our heads.
  - We’re all friends here. How would the reviewer be saying this to you in person?
- Try to respond to every comment and resolve them as you go.
  - Link to specific commits - "Well spotted! Done in 123987"
- Provide links and learning resources that you found useful so the reviewer
can also benefit from your research.
  - If you have needed to look something up, the reviewer might too.
  - Put links in the description to tutorials & articles that helped you.
  - Leave links to useful documentation as comments in the code and future devs

>   will also benefit.

Reviewer: ```suggestion
  can also benefit.
```

- Prefer to involve the person at your desk/on a Zoom call, rather than just
  communicating backwards and forwards over pull request comments. But
  ultimately you be the judge!
  - Post a comment to summarize any discussions that took place.
- Delete the branch after merging the pull request.
- Sharing knowledge avoids silos and any comment is a learning opportunity
  for both so ask questions.
- Be humble.
  - Assume the reviewee has the best understanding of the problems being
  solved, they’ve put a lot of thought and work into it.
  - Trust in their decisions but ask for clarification.
- Be aware that written feedback is often read in an overly critical tone.
  - Prefer open-ended questions over making strong or opinionated statements.
  - Prefer we instead of you.

>     - "-another strategy- would be better here."

Reviewer: ```suggestion
    - bad: "-another strategy- would be better here."
```

    - better: "We should use -another strategy- because -a reason-."
    - great: "Could we use -another strategy- for -a reason-? Why did you chose
    this way?"
- If there’s a lot of feedback:
  - Consider what are the must dos and what, if anything, can be left out to
  avoid over-facing someone & keep us progessing.
    - Must dos - the important stuff. Highlight them in a change request.
    - Should dos - make a task/ticket if anything can be done as later.
    - Could dos - "We could do -suggestion-. Not important, you decide."

> - Check the [handbook](./03-style-guidelines.md) for things that are to do with

Reviewer: ```suggestion
- Use the [handbook](./03-style-guidelines.md) as a code style reference.
```

(and delete next line)

code style.

>   - We should be on the same page with our code style, if not, prefer to discuss as

Reviewer: ```suggestion
  - We should be on the same page with our code style, if not, we should discuss as
```

  a team rather than between individuals on PR’s.
  - If it’s not in the handbook or you disagree with something, bring it up in

>   retro & then update the handbook once we have agreement.

Reviewer: ```suggestion
  retro and then update the handbook once we have agreement.
```

  - As time goes on we'll all know what's expected and these comments will
  disappear from PRs - speed bonus :D
- Do give praise for good work :)
  - Highlight things you liked
  - Highlight things you’ve learned

>   - ::taco::

Reviewer: ```suggestion
  - Give an appreciation ::taco:: in #appreciations on Slack!
```

- Rails: Does brakeman report any new security issues?
- Asset pipeline JavaScript: Have any ES6-isms snuck in e.g. const/let, arrow
  functions? This is OK if the JS is being compiled by Webpacker or similar.

> - React: Are there any missing PropTypes?

Reviewer: Maybe add a link to the proptypes library?
